compiler/rustc_ast_passes/src/feature_gate.rs

```diff
@@ -677,3 +678,36 @@ fn maybe_stage_features(sess: &Session, krate: &ast::Crate) {
         }
     }
 }
+
+fn check_incompatible_features(sess: &Session) {
+    let features = sess.features_untracked();
+
+    let declared_features = features
+        .declared_lang_features
+        .iter()
+        .map(|(name, span, _)| (name, span))
+        .chain(features.declared_lib_features.iter().map(|(name, span)| (name, span)));
+
+    for (f1, f2) in rustc_feature::INCOMPATIBLE_FEATURES
+        .iter()
+        .filter(|(f1, f2)| features.enabled(*f1) && features.enabled(*f2))
```

Comment:
```suggestion
        .filter(|&(f1, f2)| features.enabled(f1) && features.enabled(f2))
```

Reply:
For some reason, this does not work. Maybe because `rustc_feature::INCOMPATIBLE_FEATURES` is `&[..]`

Reply:
Can you show me the error message here?

Reply:
```
error[E0308]: mismatched types
   --> compiler/rustc_ast_passes/src/feature_gate.rs:694:46
    |
694 |         .filter(|&(f1, f2)| features.enabled(f1) && features.enabled(f2))
    |                                              ^^
    |                                              |
    |                                              expected struct `rustc_span::Symbol`, found `&rustc_span::Symbol`
    |                                              help: consider dereferencing the borrow: `*f1`

error[E0308]: mismatched types
   --> compiler/rustc_ast_passes/src/feature_gate.rs:694:70
    |
694 |         .filter(|&(f1, f2)| features.enabled(f1) && features.enabled(f2))
    |                                                                      ^^
    |                                                                      |
    |                                                                      expected struct `rustc_span::Symbol`, found `&rustc_span::Symbol`
    |                                                                      help: consider dereferencing the borrow: `*f2`

```

Reply:
```suggestion
        .filter(|&&(f1, f2)| features.enabled(f1) && features.enabled(f2))
```
does this work? 1 for `iter()` and one for `filter` (as it uses `&I::Item`)

Reply:
haha, actually it does. Thanks. I'll push now.